migration/sql-files/103-update-non-root-iteration-path.sql

```diff
@@ -1,3 +1,9 @@
 -- update root iteration and area paths to use converted ids
 UPDATE iterations SET path=text2ltree(concat(path, concat('.',replace(cast(id as text), '-', '_')))) WHERE path!='' OR NOT NULL; 
-UPDATE areas SET path=text2ltree(concat(path, concat('.',replace(cast(id as text), '-', '_')))) WHERE path!='' OR NOT NULL; 
\ No newline at end of file
+UPDATE areas SET path=text2ltree(concat(path, concat('.',replace(cast(id as text), '-', '_')))) WHERE path!='' OR NOT NULL; 
+
+ALTER TABLE iterations DROP CONSTRAINT iterations_name_space_id_path_unique;
+ALTER TABLE areas DROP CONSTRAINT areas_name_space_id_path_unique;
+
+CREATE UNIQUE INDEX areas_name_space_id_path_unique ON areas (space_id, subpath(path, 0, -1), name) WHERE deleted_at IS NULL;
```

Comment:
@tinakurian can you add a comment here for why we need this new index? Something like "two areas/iterations under the same parent area/iteraion must have a unique name" would be enough I think.

Also I think technically the `space_id` is not needed for this. It was needed when a root iteration had an empty path before. Back then you could have two iterations "foo" within two spaces and they both had the same path (`""` or `NULL`). This is no longer the case because both iteration have their own converted ID as their path.

I'm not sure if an index over the `space_id` would add to the performance of the index. @aslakknutsen shall we keep the `space_id` in the index?